extensions-core/druid-lookups/src/main/java/io/druid/server/lookup/jdbc/JdbcDataFetcher.java

```diff
@@ -0,0 +1,202 @@
+  private final String fetchQuery;
+  private final String reverseFetchQuery;
+
+  public JdbcDataFetcher(
+      @JsonProperty("connectorConfig") MetadataStorageConnectorConfig connectorConfig,
+      @JsonProperty("table") String table,
+      @JsonProperty("keyColumn") String keyColumn,
+      @JsonProperty("valueColumn") String valueColumn
+  )
+  {
+    this.connectorConfig = Preconditions.checkNotNull(connectorConfig, "connectorConfig");
+    Preconditions.checkNotNull(connectorConfig.getConnectURI(), "connectorConfig.connectURI");
+    this.table = Preconditions.checkNotNull(table, "table");
+    this.keyColumn = Preconditions.checkNotNull(keyColumn, "keyColumn");
+    this.valueColumn = Preconditions.checkNotNull(valueColumn, "valueColumn");
+
+    this.fetchAllQuery = String.format(
+        "SELECT %s, %s FROM %s",
+        this.keyColumn,
+        this.valueColumn,
+        this.table
+    );
+    this.fetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.valueColumn,
+        this.table,
+        this.keyColumn
+    );
+    this.reverseFetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.keyColumn,
+        this.table,
+        this.valueColumn
+    );
+  }
+
+  @Override
+  public Map<String, String> fetchAll()
+  {
+    final DBI dbi = getDbi();
+    return dbi.withHandle(new HandleCallback<ImmutableMap<String, String>>()
+                          {
+                            @Override
+                            public ImmutableMap<String, String> withHandle(Handle handle) throws Exception
+                            {
+                              final ImmutableMap.Builder mapBuilder = ImmutableMap.builder();
+                              List<Map<String, Object>> result = handle.select(fetchAllQuery);
+                              LOGGER.debug("Polled [%d] row(s) from table [%s]", result.size(), table);
+                              for (Map<String, Object> map : result) {
+                                mapBuilder.put(
+                                    map.get(keyColumn),
+                                    map.get(Strings.nullToEmpty(valueColumn))
+                                );
+                              }
+                              return mapBuilder.build();
+                            }
+                          }
+    );
+  }
+
+  @Override
+  public String fetch(final String key)
+  {
+    final DBI dbi = getDbi();
+    List<String> pairs = dbi.withHandle(
+        new HandleCallback<List<String>>()
+        {
+          @Override
+          public List<String> withHandle(Handle handle) throws Exception
+          {
+            return handle.createQuery(fetchQuery)
+                         .bind("val", key)
+                         .map(StringMapper.FIRST)
+                         .list();
+          }
+        }
+    );
+    if (pairs.isEmpty()) {
+      return null;
+    }
+    return pairs.get(0);
```

Comment:
fetch all converts values from null to empty string, same is not done here. is that consistent?